crypto/bf/bf_enc.c

```diff
@@ -26,6 +26,12 @@ void BF_encrypt(BF_LONG *data, const BF_KEY *key)
     register BF_LONG l, r;
     register const BF_LONG *p, *s;
 
+    if (FIPS_mode()) {
+        FIPSerr(ERR_LIB_FIPS, FIPS_R_NON_FIPS_METHOD);
+        OpenSSLDie(__FILE__, __LINE__, "FATAL FIPS Unapproved algorithm called");
+        return;
```

Comment:
Do we really need the return?

Reply:
Hmmm. I always use return in void cases, even after the call to OpenSSLDie() as a "belt and braces" approach. I just don't want it blundering into subsequent code for *any* reason. Judgement call - I'll go with whatever you desire on this.

Reply:
I think it makes it clear that we have no intention to execute the reset of the function and if someone didn't know exactly what `OpenSSLDie` does (yes its self explanatory) even if that `return` is never executed.